File: Lesson03/OnClickListeners/app/src/main/java/com/codingblocks/onclicklisteners/MainActivity.java
package com.codingblocks.onclicklisteners;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    Button btnIncrement, btnDecrement;
    TextView tvCount;
    Integer count = 0;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        btnDecrement = findViewById(R.id.btnDecrement);
        btnIncrement = findViewById(R.id.btnIncrement);
        tvCount = findViewById(R.id.tvCount);

//        Fruit f = new Fruit() {
//            @Override
//            public String getOrigin() {
//                return null;
//            }
//
//            @Override
//            public String getTaste() {
//                return null;
//            }
//
//            @Override
//            public String getPrice() {
//                return null;
//            }
//        };

        //Can you create an instance of an interface? No

//        View.OnClickListener clickListener = new MyClickListener();

        tvCount.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                count = 0;
                tvCount.setText(count.toString());
            }
        });

        btnIncrement.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                count++;
                //Perform the computation to increase the count
                tvCount.setText(count.toString());
            }
        });

        btnDecrement.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                count--;
                //Perform the computation to decrease the count
                tvCount.setText(count.toString());
            }
        });

    }

    public void increment(View view) {

    }

    public void decrement(View view) {

    }
}

File: Lesson07/Kotlin/app/src/main/java/com/codingblocks/kotlin/MainActivity.kt
package com.codingblocks.kotlin

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.support.v4.content.ContextCompat
import android.widget.Toast
import kotlinx.android.synthetic.main.activity_main.*
import kotlin.concurrent.thread

class MainActivity : AppCompatActivity() {

    //vars can be reassigned
    private var count: Int = 0

    //vals can't be reassigned
    val valCount: Int = 0

    lateinit var noteAdapter: NoteAdapter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val sampleText = "Hello World"

        noteAdapter = NoteAdapter()

        count++
        count--

        val titleNote = Note(title = "sdkjfdsb")

        val descNote = Note(desc = "sdfsdf")


        tvOne.setOnClickListener {

            tvTwo.setTextColor(ContextCompat.getColor(this, R.color.colorAccent))

        }


//        valCount++

        thread {
            //Do background processing here

            runOnUiThread {
                //Do the UI work here
            }
        }

    }

    fun increment() {
        count++
    }

    fun decrement() {
        count--
    }

    fun countSquared(): Int {
        return count * count
    }

    fun singleLineCountSquared() = count * count

    fun showToast(number: Int) {
        Toast.makeText(this, "The number is $number", Toast.LENGTH_SHORT).show()
    }

    fun printEvenOrOdd(number: Int) {

        val isEven = if (number % 2 == 0) true else false

        showShortToast(this, "HEllo world")

        Toast.makeText(this, "The Square of number is : ${countSquared()}", Toast.LENGTH_SHORT).show()

        Toast.makeText(
            this,
            "$number is : ${if (number % 2 == 0) "even" else "odd"}",
            Toast.LENGTH_SHORT
        ).show()
    }

}
